src/components/Images.vue

```diff
@@ -271,57 +279,158 @@ export default {
-          this.mainWindowScroll = -1;
+          this.$refs.fullWindow.parentElement.parentElement.scrollTop = this.$refs.fullWindow.scrollHeight;
         });
       }
     },
+    closeOutputWindow(event) {
+      this.keepImageManagerOutputWindowOpen = false;
+      if (this.postCloseOutputWindowHandler) {
+        this.postCloseOutputWindowHandler();
+        this.postCloseOutputWindowHandler = null;
+      } else {
+        this.imageManagerOutput = '';
+        if (this.mainWindowScroll >= 0) {
+          this.$nextTick(() => {
+            try {
+              this.$refs.fullWindow.parentElement.parentElement.scrollTop = this.mainWindowScroll;
+            } catch (e) {
+              console.log(`Trying to reset scroll to ${ this.mainWindowScroll }, got error:`, e);
+            }
+            this.mainWindowScroll = -1;
+          });
+        }
+      }
+    },
     doClick(row, rowOption) {
+      // Do this in case a handler from the previous operation didn't fire due to an error.
       rowOption.action(row);
     },
     startRunningCommand(command) {
-      this.keepImageManagerOutputWindowOpen = true;
       this.imageOutputCuller = getImageOutputCuller(command);
+    },
+    async deleteImage(obj) {
+      const options = {
+        message:   `Delete image ${ obj.imageName }:${ obj.tag }?`,
+        type:      'question',
+        buttons:   ['Yes', 'No'],
+        defaultId: 1,
+        title:     'Confirming image deletion',
+        cancelId:  1
+      };
+      const result = await ipcRenderer.invoke('show-message-box', options);
 
-      if (this.$refs.fullWindow) {
-        this.$nextTick(() => {
-          this.$refs.fullWindow.parentElement.parentElement.scrollTop = this.$refs.fullWindow.scrollHeight;
-        });
+      if (result.response === 1) {
+        return;
       }
-    },
-    deleteImage(obj) {
       this.currentCommand = `delete ${ obj.imageName }:${ obj.tag }`;
       this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
       this.startRunningCommand('delete');
-      ipcRenderer.send('confirm-do-image-deletion', obj.imageName.trim(), obj.imageID.trim());
+      ipcRenderer.send('do-image-deletion', obj.imageName.trim(), obj.imageID.trim());
     },
     doPush(obj) {
       this.currentCommand = `push ${ obj.imageName }:${ obj.tag }`;
+      this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
       this.startRunningCommand('push');
       ipcRenderer.send('do-image-push', obj.imageName.trim(), obj.imageID.trim(), obj.tag.trim());
     },
     doBuildAnImage() {
-      this.currentCommand = `build ${ this.imageToBuild }`;
+      const imageName = this.imageToBuild.trim();
+
+      this.currentCommand = `build ${ imageName }`;
       this.fieldToClear = 'imageToBuild';
+      this.postCloseOutputWindowHandler = () => this.scrollToImageOnSuccess(imageName);
```

Comment:
I had a look at this, and the code does refresh the image list after both builds and pulls.

It also doesn't make much sense to retry after a delay to process the JS job queue, because there already was a delay because nothing happens after a build/pull until the user has pressed the "Close Output Window" button, and then it looks for the image. 

My analysis is that this is very unlikely (and you were lucky to find it), and that the worst that happens is that the new image isn't highlighted.